configuration.hpp

```diff
@@ -1,234 +1,255 @@
+			}
+			catch (std::exception& e) {
+				error_msg = fmt::format("Exception - {}", e.what());
+			}
+			catch (...) {
+				error_msg = "Unknown error";
+			}
+
+			if (error_msg.size()) {
+				irods::log(LOG_ERROR, error_msg);
+			}
+			if (!retval) {
+				if (VERBOSE_CONFIGURATION_LOADING) {
+					irods::log(LOG_ERROR, fmt::format("Using default value of {} = {}", key, default_v));
+				}
+				return default_v;
+			}
+			return *retval;
+		}
+
+	} //namespace configuration_parameters
+
+	namespace policy
+	{
+		// Policy Naming Examples
+		// irods_policy_<namespace>_<subject>_<operation>_<technology>
+		// irods_policy_indexing_object_index_
+		// irods_policy_indexing_collection_index_
+		// irods_policy_indexing_metadata_index_
+		// irods_policy_indexing_object_purge_
+		// irods_policy_indexing_collection_purge_
+		// irods_policy_indexing_metadata_purge_
+
+		static constexpr auto prefix = "irods_policy_indexing";
+		std::string compose_policy_name(const std::string& _prefix, const std::string& _technology);
+
+		namespace object
+		{
+			static const std::string index{"irods_policy_indexing_object_index"};
+			static const std::string purge{"irods_policy_indexing_object_purge"};
+		} // namespace object
+
+		namespace metadata
+		{
+			static const std::string index{"irods_policy_indexing_metadata_index"};
+			static const std::string purge{"irods_policy_indexing_metadata_purge"};
+		} // namespace metadata
+
+		namespace collection
+		{
+			static const std::string index{"irods_policy_indexing_collection_index"};
+			static const std::string purge{"irods_policy_indexing_collection_purge"};
+		} // namespace collection
+	}     // namespace policy
+
+	std::string operation_and_index_types_to_policy_name(const std::string& _operation_type,
+	                                                     const std::string& _index_type);
+
+	namespace schedule
+	{
+		static const std::string object{"irods_policy_schedule_object_indexing"};
+		static const std::string collection{"irods_policy_schedule_collection_indexing"};
+	} // namespace schedule
+
+	namespace index_type
+	{
+		static const std::string full_text{"full_text"};
+		static const std::string metadata{"metadata"};
+	} // namespace index_type
+
+	namespace operation_type
+	{
+		static const std::string index{"index"};
+		static const std::string purge{"purge"};
+	} // namespace operation_type
+
+	struct configuration
+	{
+		// metadata attributes
+		std::string index{"irods::indexing::index"};
+		std::string flag{"irods::indexing::flag"};
+
+		// basic configuration
+		int minimum_delay_time{1};
+		int maximum_delay_time{30};
+		int job_limit{};
+		std::string delay_parameters{"<EF>60s DOUBLE UNTIL SUCCESS OR 5 TIMES</EF>"};
+
+		// Clients should aim not to use this index-embedded URL. It is no
+		// longer used by MetaLnx as of version 2.5.0, nor by the search plugin
+		// extension. Thus, it is a possible target for deprecation.
+		std::string url_template{"http:/{}"};
```

Comment:
let's remove/deprecate this - years have passed.

Reply:
obviously, separate PR is fine fine.

Reply:
Done.